### PR TITLE
[pt1][quant][annotation] Expose QScheme in frontend

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#19381 [pt1][quant][annotation] Expose QScheme in frontend**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D14922992/)

Expose QScheme enum in frontend so that people can use it in
quantization configs in modules.

Differential Revision: [D14922992](https://our.internmc.facebook.com/intern/diff/D14922992/)

### CODE REVIEW

File: torch/nn/_qschme.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from .._jit_internal import weak_script
```
Comment (dzhulgakov) [inline torch/nn/_qschme.py]: the name of this entire file is misspelled (_qschme)

**Review by dzhulgakov — APPROVED** [review]

File: torch/nn/_qschme.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    elif qscheme == 'per_tensor_affine':
+        ret = 1
+    elif qscheme == 'per_channel_affine':
+        ret = 2
+    elif qscheme == 'per_tensor_symmetric':
+        ret = 3
+    elif qscheme == 'per_channel_symmetric':
+        ret = 4
+    else:
+        ret = -1  # TODO: remove once JIT exceptions support control flow
```
Comment (ZolotukhinM) [inline torch/nn/_qschme.py]: You probably want to either return -1 or raise an error - right now assigning -1 is useless since we're throwing on the next line and never return -1.

Comment (gchanan) [reply torch/nn/_qschme.py]: I'm not sure exactly what is happening here.

**Review by gchanan — COMMENTED** [review]
are you going to hook this up in a future PR?

File: aten/src/ATen/core/Reduction.h [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 namespace Reduction {
 
-// NB: Keep this in sync with Reduction class in torch/nn/modules/functional.py
+// NB: Keep this in sync with Reduction class in torch/nn/_reduction.py
```
Comment (ZolotukhinM) [inline aten/src/ATen/core/Reduction.h]: Is this a related change? If not, please split it.

File: torch/nn/_qschme.py (hunk 2 above)
Comment (jerryzh168) [reply torch/nn/_qschme.py]: this is copied from _reduction.py

Comment (jerryzh168) [reply torch/nn/_qschme.py]: I'll remove ret = -1, what does the comment mean?

File: aten/src/ATen/core/Reduction.h (hunk 3 above)
Comment (jerryzh168) [reply aten/src/ATen/core/Reduction.h]: not related to this but can we just put it here? since it's just a small comment change.

File: torch/nn/_qschme.py (hunk 2 above)
Comment (ZolotukhinM) [reply torch/nn/_qschme.py]: I think there is enough support in JIT for it now - we can script code throwing exceptions. Maybe it was written when it was not the case, and the author meant to keep the raise statement commented out.
